codegen/src/utils/fetch_metadata.rs

```diff
@@ -32,26 +49,80 @@ fn tokio_block_on<T, Fut: std::future::Future<Output = T>>(fut: Fut) -> T {
 }
 
 /// Returns the metadata bytes from the provided URL.
-pub async fn fetch_metadata_bytes(url: &Uri) -> Result<Vec<u8>, FetchMetadataError> {
-    let hex = fetch_metadata_hex(url).await?;
-    let bytes = hex::decode(hex.trim_start_matches("0x"))?;
-    Ok(bytes)
-}
-
-/// Returns the raw, 0x prefixed metadata hex from the provided URL.
-pub async fn fetch_metadata_hex(url: &Uri) -> Result<String, FetchMetadataError> {
-    let hex_data = match url.scheme_str() {
-        Some("http") | Some("https") => fetch_metadata_http(url).await,
-        Some("ws") | Some("wss") => fetch_metadata_ws(url).await,
+pub async fn fetch_metadata_bytes(
+    url: &Uri,
+    version: MetadataVersion,
+) -> Result<Vec<u8>, FetchMetadataError> {
+    let bytes = match url.scheme_str() {
+        Some("http") | Some("https") => fetch_metadata_http(url, version).await,
+        Some("ws") | Some("wss") => fetch_metadata_ws(url, version).await,
         invalid_scheme => {
             let scheme = invalid_scheme.unwrap_or("no scheme");
             Err(FetchMetadataError::InvalidScheme(scheme.to_owned()))
         }
     }?;
+
+    Ok(bytes)
+}
+
+/// Returns the raw, 0x prefixed metadata hex from the provided URL.
+pub async fn fetch_metadata_hex(
+    url: &Uri,
+    version: MetadataVersion,
+) -> Result<String, FetchMetadataError> {
+    let bytes = fetch_metadata_bytes(url, version).await?;
+    let hex_data = format!("0x{}", hex::encode(bytes));
     Ok(hex_data)
 }
 
-async fn fetch_metadata_ws(url: &Uri) -> Result<String, FetchMetadataError> {
+/// Execute runtime API call and return the specified runtime metadata version.
+async fn fetch_metadata(
+    client: impl ClientT,
+    version: MetadataVersion,
+) -> Result<Vec<u8>, FetchMetadataError> {
+    const UNSTABLE_METADATA_VERSION: u32 = u32::MAX;
+    const LATEST_STABLE_VERSION: u32 = 14;
```

Comment:
Instead of a constant, it'd be very cool if we made a runtime call to get the list of supported versions from the node and used that to know what the latest stable version was, since maybe there will be a mix of nodes with stable V14 and stable V15 for a while, and of course new versions eventually too